doc/dev/index.rst

```diff
@@ -210,50 +227,97 @@ workflows.
+                | review                             | git commit
+                |                                    | 
                 |                                    v
            /--------------\                        /-------------\
            |   github     |<---------------------- | ceph/fix_1  |
            | pull request |         git push       \-------------/
            \--------------/
 
+We will now walk through this chart from the point of view of a beginning
+developer who has an idea for a bugfix, but does not know exactly how to
+proceed.
 
-Issue tracker conventions
--------------------------
+Before you start, you should know the `issue tracker`_ number of the bug
+you intend to fix. If there is no tracker issue, create one. If you have
+sufficient tracker permissions, assign the bug to yourself by changing the
+`Assignee` field. If your tracker permissions have not yet been elevated,
+simply update the issue with a short note (in English) to let the other
+developers know you are working on the bug.
 
-When you start working on an existing issue, it's nice to let the other
-developers know this - to avoid duplication of labor. Typically, this is
-done by changing the :code:`Assignee` field (to yourself) and changing the
-:code:`Status` to *In progress*. Newcomers to the Ceph community typically do not
-have sufficient privileges to update these fields, however: they can
-simply update the issue with a brief note.
+The following sections correspond to the nodes in the above chart.
 
-.. table:: Meanings of some commonly used statuses
+Upstream code
+-------------
 
-   ================ ===========================================
-   Status           Meaning
-   ================ ===========================================
-   New              Initial status
-   In Progress      Somebody is working on it
-   Need Review      Pull request is open with a fix
-   Pending Backport Fix has been merged, backport(s) pending
-   Resolved         Fix and backports (if any) have been merged
-   ================ ===========================================
+This lives in https://github.com/ceph/ceph and is the starting and ending
+point of our odyssey.
 
-Pull requests
+A local copy of the upstream code is made by forking `ceph/ceph.git` on
+GitHub and using `git clone` to make a local copy of the fork.
+
+Local environment
+-----------------
+
+In the local environment created in the previous step, we now have a local
+copy of the `master` branch in `origin/master`. Since the fork is frozen in
+time and `ceph/ceph.git` is updated frequently by other develoeprs, we will
+need to add `ceph/ceph.git` as a remote so we can fetch from it:
+
+.. code::
+
+    $ git remote add ceph https://github.com/ceph/ceph.git
+    $ git fetch ceph
+
+After running these commands, we have a branch called `ceph/master` in our
+local tree. Our local `master` branch (and that of our fork on GitHub) can
```

Comment:
local tree => local git clone ? The ceph/master branch is not in the working tree until you git checkout -b foobar ceph/master in which case it is in the local foobar branch as well as the working tree.